<?xml version="1.0"?>
<launch>

    <param name="scan_period" value="0.5" />    
    <param name="map_resolution" type="double" value="1" />
    <param name="max_dis" type="double" value="25.0" />
    <param name="min_dis" type="double" value="0" />
    <param name="vertical_angle" type="double" value="0" />
    <param name="validation_angle" type="bool" value="false" />

    <!--- --> 
    <node pkg="lilo" type="odom_estimation_node" name="odom_estimation_node" output="screen"/>
    <node pkg="lilo" type="laser_processing_node" name="laser_processing_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />
    
    <!--- -->
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find lilo)/rviz/lilo_velodyne.rviz" />
    </group>
    <!--- -->

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
